src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -491,4 +495,202 @@ private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOExcep
+            System.out.println("Checkout: " + co.toString());
+            assertTrue(co.pollingBaseline != null);
+        }
+
+        p.setDefinition(new CpsFlowDefinition("semaphore 'waitFirst'\n" +
+                "node {\n" +
+                "    checkout(changelog:" + false +", poll:" + true + ", scm: [$class: 'GitSCM', branches: [[name: '*/master']], " +
+                "doGenerateSubmoduleConfigurations: false, extensions: [], " +
+                "gitTool: 'Default', submoduleCfg: [], userRemoteConfigs: [[url: '" + sampleRepo + "']]])" +
+                "}\n", false /* for org.jvnet.hudson.test.FakeChangeLogSCM */));
+
+
+        WorkflowRun b2 = p.scheduleBuild2(0).waitForStart();
+
+        SemaphoreStep.waitForStart("waitFirst/2", b2);
+        SemaphoreStep.success("waitFirst/2", null);
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b2));
+        for (WorkflowRun.SCMCheckout co : b2.checkouts(listener)){
+            assertTrue(co.pollingBaseline != null);
+        }
+        p.setDefinition(new CpsFlowDefinition("semaphore 'waitFirst'\n" +
+                "node {\n" +
+                "    checkout(changelog:" + true +", poll:" + false + ", scm: [$class: 'GitSCM', branches: [[name: '*/master']], " +
+                "doGenerateSubmoduleConfigurations: false, extensions: [], " +
+                "gitTool: 'Default', submoduleCfg: [], userRemoteConfigs: [[url: '" + sampleRepo + "']]])" +
+                "}\n", false /* for org.jvnet.hudson.test.FakeChangeLogSCM */));
+
+
+        WorkflowRun b3 = p.scheduleBuild2(0).waitForStart();
+
+        SemaphoreStep.waitForStart("waitFirst/3", b3);
+        SemaphoreStep.success("waitFirst/3", null);
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b3));
+        for (WorkflowRun.SCMCheckout co : b3.checkouts(listener)){
+            assertTrue(co.pollingBaseline != null);
+        }
+        p.setDefinition(new CpsFlowDefinition("semaphore 'waitFirst'\n" +
+                "node {\n" +
+                "    checkout(changelog:" + false +", poll:" + false + ", scm: [$class: 'GitSCM', branches: [[name: '*/master']], " +
+                "doGenerateSubmoduleConfigurations: false, extensions: [], " +
+                "gitTool: 'Default', submoduleCfg: [], userRemoteConfigs: [[url: '" + sampleRepo + "']]])" +
+                "}\n", false /* for org.jvnet.hudson.test.FakeChangeLogSCM */));
+
+
+        WorkflowRun b4 = p.scheduleBuild2(0).waitForStart();
+
+        SemaphoreStep.waitForStart("waitFirst/4", b4);
+        SemaphoreStep.success("waitFirst/4", null);
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b4));
+        for (WorkflowRun.SCMCheckout co : b4.checkouts(listener)){
+            assertTrue(co.pollingBaseline == null);
+        }
+    }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Basic test to prove that checkout gets individually set for each checkout
+    public void baselineResetMultipleRepos() throws Exception {
```

Comment:
What happens if you checkout the same repo twice in one pipeline with different settings?